stan/math/prim/fun/spaced_array.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef STAN_MATH_PRIM_FUN_SPACED_ARRAY_HPP
+#define STAN_MATH_PRIM_FUN_SPACED_ARRAY_HPP
+
+#include <stan/math/prim/err.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return an array of linearly spaced elements
+ *
+ * This produces an array from low to high (included) with elements spaced
+ * as (high - low) / (K - 1). For K=1, the arry will contain the high value;
+ * for K=0 it returns an empty array.
+ *
+ * @param K size of the array
+ * @param low smallest value
+ * @param high largest value
+ * @return An array of size K with elements linearly spaced between
+ * low and high.
+ * @throw std::domain_error if K is negative, if low is nan or infinite,
+ * if high is nan or infinite, or if high is less than low.
+ */
+inline std::vector<double> spaced_array(int K, double low, double high) {
+  static const char* function = "spaced_array";
+  check_nonnegative(function, "size", K);
+  check_finite(function, "low", low);
+  check_finite(function, "high", high);
+  check_greater_or_equal(function, "high", high, low);
+
+  if (K == 0) {
```

Comment:
[optional]

This could be implemented as

```
Eigen::VectorXd a = Eigen::VectorXd::LinSpaced(K, low, high);
return {&a[0], &a[0] + K};
```

It's an extra copy but a lot less code and ensures we get the same answers in all of our functions.
It should match the input iterator constructor.

Reply:
This works, but the `K == 0` case has to be kept separate.